tests/units/TicketTest.php

```diff
@@ -359,4 +360,393 @@ public function testEscalateButtonShouldTriggerGroupEscalationAndExecuteRuleOnTi
         $this->assertEquals(1, count($group_ticket->find(['tickets_id' => $ticket->getID(), 'groups_id' => $group_tech_id, 'type' => \CommonITILActor::ASSIGN])));
         $this->assertEquals(1, count($group_ticket->find(['tickets_id' => $ticket->getID(), 'groups_id' => $group_observer_id, 'type' => \CommonITILActor::OBSERVER])));
     }
+
+    private function testAssignGroupToTicketWithCategoryProvider()
+    {
+        yield [
+            'conf' => [
+                'remove_tech'             => 0,
+                'remove_group'            => 0,
+                'reassign_group_from_cat' => 0,
+                'reassign_tech_from_cat'  => 0,
+            ],
+            'expected' => [
+                'users' => 1,
+                'groups' => 1,
+                'user_1_is_assign' => 1,
+                'user_2_is_assign' => 0,
+                'group_1_is_assign' => 1,
+                'group_2_is_assign' => 0,
+            ],
```

Comment:
users is not necessary, as it amounts to counting user_1_is_assign + user_2_is_assign. The same applies to groups. You can lighten the test by removing them.
```suggestion
            'expected' => [
                'user_1_is_assign' => 1,
                'user_2_is_assign' => 0,
                'group_1_is_assign' => 1,
                'group_2_is_assign' => 0,
            ],
```